core/bugnote_api.php

```diff
@@ -753,10 +753,38 @@ function bugnote_clear_cache( $p_bugnote_id = null ) {
 
 	if( null === $p_bugnote_id ) {
 		$g_cache_bugnote = array();
+		$g_cache_bugnotes = array();
 	} else {
-		unset( $g_cache_bugnote[(int)$p_bugnote_id] );
+		if( isset( $g_cache_bugnote[(int)$p_bugnote_id] ) ) {
+			$t_note_obj = $g_cache_bugnote[(int)$p_bugnote_id];
+			bugnote_clear_bug_cache( $t_note_obj->bug_id );
+			unset( $g_cache_bugnote[(int)$p_bugnote_id] );
+		}
 	}
-	$g_cache_bugnotes = array();
 
 	return true;
 }
+
+/**
+ * Clear the bugnotes related to a bug, or all bugs if no bug id specified.
+ * @param integer $p_bug_id Identifier to clear (optional).
+ * @return boolean
+ * @access public
+ */
+function bugnote_clear_bug_cache( $p_bug_id = null ) {
+	global $g_cache_bugnotes, $g_cache_bugnote;
+
+	if( null === $p_bug_id ) {
+		$g_cache_bugnotes = array();
+		$g_cache_bugnote = array();
```

Comment:
I never noticed we had 2 distinct variables for bugnotes caching. It is not really clear what the difference between them are, and they are not formally declared at the top of bugnote api, only as global in the functions using them. Do we actually need 2 of them ?

Even though that's not directly related to this PR, I would suggest to declare them, and add a PHPDoc block to explain their purpose and differences.